packages/hint-manifest-icons/tests/tests.ts

```diff
@@ -0,0 +1,204 @@
+import * as fs from 'fs';
+
+import { test } from '@hint/utils';
+import { HintTest, testHint } from '@hint/utils-tests-helpers';
+
+const { generateHTMLPage, getHintPath } = test;
+const hintPath = getHintPath(__filename);
+
+const htmlWithManifestSpecified = generateHTMLPage('<link rel="manifest" href="site.webmanifest">');
+
+const icon192px = fs.readFileSync(`${__dirname}/fixtures/icon-192x192.png`); // eslint-disable-line no-sync
+const icon512px = fs.readFileSync(`${__dirname}/fixtures/icon-512x512.png`); // eslint-disable-line no-sync
+const icon128px = fs.readFileSync(`${__dirname}/fixtures/icon-128x128.png`); // eslint-disable-line no-sync
+
+const generateImageData = (content: Buffer): Object => {
+    return {
+        content,
+        headers: { 'Content-Type': 'image/png' }
+    };
+};
+
+const tests: HintTest[] = [
+    {
+        name: 'Web app manifest is specified with empty icons property',
+        reports: [{ message: `Valid icons property was not found in the web app manifest` }],
+        serverConfig: {
+            '/': htmlWithManifestSpecified,
+            '/site.webmanifest': {
+                content: `{
+                    "icons": []
+                }`
+            }
+        }
+    },
+    {
+        name: 'Inaccessible icon URL in the Web app manifest',
+        reports: [
+            {
+                message: `Icon could not be fetched (status code: 404).`,
+                position: {
+                    column: 29,
+                    line: 3
+                }
```

Comment:
To make test positions a bit easier to understand, we prefer the `"match"` format that provides a substring whose start should match the desired position. You can see an example of this in the tests for `hint-manifest-is-valid` (sorry I didn't realize `hint-manifest-app-name` is missing positions in its tests):

```js
        reports: [
            {
                message: `'root' should NOT have additional properties. Additional property found 'additionalProperty'.`,
                position: { match: 'additionalProperty' }
            }
        ]
```

Reply:
@antross This worked for the top level properties like `icons` but doesn't work for deeper properties like `src` or `type` **under** `icons` because the value of `position.match` is [checked with](https://github.com/webhintio/hint/blob/f934c5af80fedbdedc300597db126bbeb4df9016/packages/utils-tests-helpers/src/hint-runner.ts#L48) `indexOf` which returns `-1` for deeper checks. 

Is there a way to specify `match` for deeper properties?

Reply:
Specified the `match` without the preceding `"` like `match: 'src": "an-inaccessible-path.png"'` to make it work.
```js
{
        name: 'Inaccessible icon URL in the Web app manifest',
        reports: [
            {
                message: `Icon could not be fetched (status code: 404).`,
                position: { match: 'src": "an-inaccessible-path.png"' }
            }
        ],
        serverConfig: {
            '/': htmlWithManifestSpecified,
            '/fixtures/icon-192x192.png': generateImageData(icon192px),
            '/fixtures/icon-512x512.png': generateImageData(icon512px),
            '/site.webmanifest': {
                content: `{
                    "icons": [
                        {
                            "src": "an-inaccessible-path.png",
                            "sizes": "152x152",
                            "type": "image/png"
                        }
                    ]
                }`
            }
        }
    },
```

Is there a more readable way to specify the `match`?

Reply:
Nope. That's the best approach given the location we return is deliberately the first character of the name (just past the quote). You can include the trailing quote if you want as that won't hurt anything.

FYI The position is after the first quote to ensure we underline the full name in our VS Code extension since we currently only provide a start location. If we update in the future to provide a start and end range instead we may revisit this and include the opening quote, making the match text slightly less awkward.